File: sdkwork-app-sdk-java/src/main/java/com/sdkwork/app/model/RankingVO.java
package com.sdkwork.app.model;

public class RankingVO {
    private String createdAt;
    private String updatedAt;
    private String rankingType;
    private String period;
    private List<RankInfoVO> rankings;
    private RankInfoVO myRank;
    private Integer totalUsers;

    public String getCreatedAt() {
        return this.createdAt;
    }
    
    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt() {
        return this.updatedAt;
    }
    
    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getRankingType() {
        return this.rankingType;
    }
    
    public void setRankingType(String rankingType) {
        this.rankingType = rankingType;
    }

    public String getPeriod() {
        return this.period;
    }
    
    public void setPeriod(String period) {
        this.period = period;
    }

    public List<RankInfoVO> getRankings() {
        return this.rankings;
    }
    
    public void setRankings(List<RankInfoVO> rankings) {
        this.rankings = rankings;
    }

    public RankInfoVO getMyRank() {
        return this.myRank;
    }
    
    public void setMyRank(RankInfoVO myRank) {
        this.myRank = myRank;
    }

    public Integer getTotalUsers() {
        return this.totalUsers;
    }
    
    public void setTotalUsers(Integer totalUsers) {
        this.totalUsers = totalUsers;
    }
}

File: sdkwork-app-sdk-typescript/src/types/ranking-vo.ts
import type { RankInfoVO } from './rank-info-vo';

/** 排行榜 */
export interface RankingVO {
  /** 创建时间 */
  createdAt?: string;
  /** 更新时间 */
  updatedAt?: string;
  /** 排行榜类型 */
  rankingType?: string;
  /** 周期 */
  period?: string;
  /** 排名列表 */
  rankings?: RankInfoVO[];
  /** 我的排名 */
  myRank?: RankInfoVO;
  /** 总用户数 */
  totalUsers?: number;
}
